mne/viz/topo.py

```diff
@@ -76,68 +76,205 @@ def iter_topography(info, layout=None, on_pick=None, fig=None,
     if fig is None:
         fig = plt.figure()
 
-    fig.set_facecolor(fig_facecolor)
-    if layout is None:
-        from ..channels import find_layout
-        layout = find_layout(info)
-
     if on_pick is not None:
         callback = partial(_plot_topo_onpick, show_func=on_pick)
         fig.canvas.mpl_connect('button_press_event', callback)
 
-    pos = layout.pos.copy()
-    if layout_scale:
-        pos[:, :2] *= layout_scale
+    fig.set_facecolor(fig_facecolor)
 
     ch_names = _clean_names(info['ch_names'])
-    iter_ch = [(x, y) for x, y in enumerate(layout.names) if y in ch_names]
-    for idx, name in iter_ch:
+    ch_types = dict((ch_name, channel_type(info, j))
+                    for j, ch_name in enumerate(ch_names))
+
+    iter_ch = [(x, y) for x, y in enumerate(info["ch_names"]) if y in ch_names]
+    for idx, ch_name in iter_ch:
         ax = plt.axes(pos[idx])
-        ax.patch.set_facecolor(axis_facecolor)
-        plt.setp(list(ax.spines.values()), color=axis_spinecolor)
-        ax.set_xticklabels([])
-        ax.set_yticklabels([])
-        plt.setp(ax.get_xticklines(), visible=False)
-        plt.setp(ax.get_yticklines(), visible=False)
-        ch_idx = ch_names.index(name)
-        vars(ax)['_mne_ch_name'] = name
+#        plt.setp(list(ax.spines.values()), color=axis_spinecolor)
+#        ax.set_xticklabels([])
+#        ax.set_yticklabels([])
+#        plt.setp(ax.get_xticklines(), visible=True)
+#        plt.setp(ax.get_yticklines(), visible=True)
+        ch_idx = ch_names.index(ch_name)
+#        ch_type = ch_types[ch_name]
+        vars(ax)['_mne_ch_name'] = ch_name
         vars(ax)['_mne_ch_idx'] = ch_idx
-        vars(ax)['_mne_ax_face_color'] = axis_facecolor
+        vars(ax)['_mne_ch_type'] = ch_types[ch_name]
+        vars(ax)['_mne_ax_face_color'] = 'w'
         yield ax, ch_idx
 
 
+def _scale_layout(layout, layout_scale):
+
+    pos = layout.pos.copy()
+
+    if layout_scale is 'auto':
+        from itertools import combinations
+        layout_scale = 0.1
+        pos[:, :2] *= layout_scale
+
+        def _area(a, b):
+            dx = min(a[0] + a[2], b[0] + b[2]) - max(a[0], b[0])
+            dy = min(a[1] + a[3], b[1] + b[3]) - max(a[1], b[1])
+            return ((dx >= 0) and (dy >= 0))
+
+        while any(_area(a, b) for a, b in combinations(pos, 2)):
+            # print(layout_scale)
+            layout_scale += 0.05
+            pos = layout.pos.copy()
+            pos[:, :2] *= layout_scale
+
+    else:
+        pos[:, :2] *= layout_scale
```

Comment:
you have repeated L113 here. I suppose you don't need it both in the `if` and the `else` clause?